libs/community/langchain_community/document_loaders/parsers/audio.py

```diff
@@ -12,14 +13,273 @@
+        self.byte_limit = byte_limit
+        self.max_retries = max_retries
+        self.input_format = input_format
+
+    @property
+    def _create_params(self) -> Dict[str, Any]:
+        params = {
+            "language": self.language,
+            "prompt": self.prompt,
+            "response_format": self.response_format,
+            "temperature": self.temperature,
+        }
+        return {k: v for k, v in params.items() if v is not None}
+
+    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
+        """
+        Lazily parse the provided audio blob for transcription.
+
+        Args:
+            blob (Blob):
+                The audio file in Blob format to be transcribed.
+
+        Yields:
+            Document:
+                Parsed transcription from the audio file.
+
+        Raises:
+            ImportError:
+                If the required packages `openai` or `pydub` are not installed.
+            ValueError:
+                If the file type is unsupported or exceeds the size limit.
+        """
+
+        try:
+            import openai
+        except ImportError:
+            raise ImportError(
+                "openai package not found, please install it with "
+                "`pip install openai`"
+            )
+
+        try:
+            from pydub import AudioSegment
+        except ImportError:
+            raise ImportError(
+                "pydub package not found, please install it with " "`pip install pydub`"
+            )
+
+        if not str(blob.path).endswith(self.input_format):
+            raise ValueError(
+                "File must be of one of the following types: "
+                f"{str(self.input_format)}"
+            )
+
+        if os.path.getsize(str(blob.source)) > self.byte_limit:
+            raise ValueError(
+                "Audio file exceeds 25 MB limit,"
+                f" length of file is {os.path.getsize(str(blob.source))} bytes"
+            )
+
+        if is_openai_v1():
+            client = openai.AzureOpenAI(
+                api_key=self.api_key,
+                azure_endpoint=self.azure_endpoint,
+                api_version=self.api_version,
+                max_retries=self.max_retries,
+                azure_ad_token=self.azure_ad_token,
+            )
+        else:
+            # Set the API key if provided
+            if self.api_key:
+                openai.api_key = self.api_key
+            if self.azure_endpoint:
+                openai.base_url = self.azure_endpoint
+
+        # Audio file from disk
+        audio = AudioSegment.from_file(blob.path)
+        # Get file extension from the file path
+        dot_file_extension = os.path.splitext(str(blob.path))[1]
+        file_extension = dot_file_extension[1:]
+        # m4a and mpga are not valid format flags for ffmpeg
+        # They do have equivalent flags that can be used though
+        format_map = {"m4a": "mp4", "mpga": "mp3"}
```

Comment:
does the service raise an exception for these formats?

Reply:
This actually had to do with pythons `io` library, it doesn't support these formats

```python
        format_map = {"m4a": "mp4", "mpga": "mp3"}
        file_extension = format_map.get(file_extension, file_extension)

        file_obj = io.BytesIO(audio.export(format=file_extension).read())
```
But I realised since we're not chunking larger files anymore and letting the API throw an error if the file is too large, this whole chunk can be taken out, it's now:

```python
        file_obj = open(blob.path, 'rb')

        #Transcribe
        try:
            if is_openai_v1():
                transcript = client.audio.transcriptions.create(
                    model=self.deployment_name,
                    file=file_obj,
                    **self._create_params,
                )
            else:
                transcript = openai.Audio.transcribe(self.deployment_name, file_obj)
        except Exception:
            raise
```